pkg/apis/kops/instancegroup.go

```diff
@@ -164,6 +164,8 @@ type InstanceGroupSpec struct {
 	// InstanceInterruptionBehavior defines if a spot instance should be terminated, hibernated,
 	// or stopped after interruption
 	InstanceInterruptionBehavior *string `json:"instanceInterruptionBehavior,omitempty"`
+	// CompressUserDataStructs gzips and base64 encodes the structs in the nodeup.sh user-data
+	CompressUserDataStructs *bool `json:"compressUserDataStructs,omitempty"`
```

Comment:
Would you mind changing to `CompressUserData`. The comment explains pretty well that only part of the user data.

```suggestion
	// CompressUserData compresses parts of the user data to save space
	CompressUserData *bool `json:"compressUserDataStructs,omitempty"`
```